### ISSUE TITLE
Include `add_qubit` and `remove_qubit` method within `EGraph` 🪄⚛️

### ISSUE THREAD

**Issue by BestQuark** [XanaduAI/flamingpy#73]
_This issue has been tagged for a bounty during [unitaryHACK](https://unitaryhack.dev/projects/flamingpy/)_.

### Task

Integrate a simple but important feature to the `EGraph` class: adding and removing qubits (nodes) of an `EGraph` object by preserving other arguments of the object consistent. This would correspond to adding or removing a separable or entangled state to the graph state (depending on whether the new node is connected by an edge to an old node). Another way to view this is as an "injection" or "deletion" operation on the graph.

You should:
- Include an `add_qubit` method to the `EGraph` class that has at least two optional arguments: `existing_neighbours` and `new_node_position`. The first of these will specify which existing qubits to entangle with the new qubit. If not supplied, the qubit will be added as a separable state. The second of these allows the user to customize where to place the qubit. If not supplied, a default convenient position should be selected (that doesn't accidentally place it too close to or at the location of another qubit).
- Include a `remove_qubit` function to the `EGraph` class that has one mandatory argument (which qubit to remove).

### Explanation

**Consider the following**:

If we draw the adjacency matrix, then add a qubit, and then draw the adjacency matrix again, we do not get an updated adj matrix.

```python
from flamingpy.utils import graph_states
state = graph_states.ring_graph(3)
state.draw_adj()
```
![image](https://user-images.githubusercontent.com/52287586/171470993-96e3fdad-2758-4ef7-9b5e-684088c310e5.png)

```python
state.add_node((2,2,2))
state.draw_adj()
```
![image](https://user-images.githubusercontent.com/52287586/171471004-17107695-9e3c-4415-accc-472895bf212d.png)

If we add a qubit, draw the adjacency matrix, then remove that qubit, and draw the adjacency matrix again, it does not get updated:

```python
state = graph_states.ring_graph(3)
state.add_node((2,2,2))
state.draw_adj()
```
![image](https://user-images.githubusercontent.com/52287586/171471565-9387c78c-464f-4643-b178-44d0e1bb1672.png)

```python
state.remove_node( (2,2,2))
state.draw_adj()
```
![image](https://user-images.githubusercontent.com/52287586/171471645-a7872929-78c7-4c3d-b127-0cf5613f34f3.png)

### Requirements

- You should add corresponding unit tests to check your modifications are covered.
- Qubits (EGraph nodes) are labelled with tuples (x, y, z).
- Other issues may arise with `EGraph.index_generator()`, `EGraph.adj_generator()`, and other functions of the `EGraph` class. The goal of these `add_qubit` and `remove_qubit` functions is to add and remove qubits effortlessly.
- It is not necessarily a good idea to regenerate indices every time a new qubit is added: it is probably enough just to add to or remove from the existing index dictionaries.

**Comment by Avhijit-codeboy** [comment]
I would like to work on this issue.

**Comment by ilan-tz** [comment]
Thanks, @Avhijit-codeboy ! Go right ahead.

**Comment by BestQuark** [comment]
I'll close this issue as it was resolved in PR #100.
